src/components/drawer/drawer.pw.tsx

```diff
@@ -0,0 +1,488 @@
+          await expect(
+            drawerSidebarContentInnerElement(page, 0)
+          ).not.toBeVisible();
+          await expect(
+            drawerSidebarContentInnerElement(page, 1)
+          ).not.toBeVisible();
+          await expect(
+            drawerSidebarContentInnerElement(page, 2)
+          ).not.toBeVisible();
+        }
+      });
+    }
+  );
+
+  ["3s", "15s"].forEach((animationDuration) => {
+    test(`should check animation time is set to ${animationDuration}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(
+        <DrawerCustom showControls animationDuration={animationDuration} />
+      );
+
+      await drawerToggle(page).click();
+      await expect(drawerSidebar(page)).toBeVisible();
+      await expect(drawerSidebar(page)).toHaveCSS(
+        "animation-duration",
+        animationDuration
+      );
+    });
+  });
+
+  test("should render component closed by default", async ({ mount, page }) => {
+    await mount(<DrawerCustom defaultExpanded={false} />);
+
+    await expect(drawerSidebarContentInnerElement(page, 0)).toHaveText(
+      "link a"
+    );
+    await expect(drawerSidebarContentInnerElement(page, 0)).not.toBeVisible();
+  });
+
+  test("should render component opened when the expanded prop is true", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<DrawerCustom expanded />);
+
+    await expect(drawerSidebarContentInnerElement(page, 0)).toHaveText(
+      "link a"
+    );
+    await expect(drawerSidebarContentInnerElement(page, 0)).toBeVisible();
+  });
+
+  test("should render component with custom sidebar", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<DrawerCustomSidebar />);
+
+    await expect(drawerSidebarContentInnerElement(page, 0)).toHaveText(
+      "cypress"
+    );
+    await expect(drawerSidebarContentInnerElement(page, 0)).toBeVisible();
+  });
+
+  test("should render component with custom expandedWidth", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<DrawerCustom expandedWidth="65%" />);
+
+    await assertCssValueIsApproximately(
+      drawer(page).locator("div").nth(0),
+      "width",
+      887
+    );
+  });
+
+  test("should render component with custom backgroundColor", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<DrawerCustom backgroundColor="#FF0000" />);
+
+    const color = "rgb(255, 0, 0)";
+
+    await expect(drawer(page).locator("div").nth(0)).toHaveCSS(
+      "background-color",
+      color
+    );
+    await expect(drawer(page).locator("div").nth(0)).toHaveCSS(
+      "border-right-color",
+      color
+    );
+  });
+
+  test("should render component with custom height", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<DrawerCustom height="75%" />);
+
+    await expect(drawer(page)).toHaveAttribute("height", "75%");
+  });
+
+  test("should render component with custom title", async ({ mount, page }) => {
+    await mount(<DrawerCustom title="cypress_title" />);
```

Comment:
**nitpick** (non-blocking): again I think it's better not to use the word "cypress" in the custom title now